src/chess/uciloop.cc

```diff
@@ -247,10 +250,21 @@ void UciLoop::SendInfo(const std::vector<ThinkingInfo>& infos) {
     if (info.seldepth >= 0) res += " seldepth " + std::to_string(info.seldepth);
     if (info.time >= 0) res += " time " + std::to_string(info.time);
     if (info.nodes >= 0) res += " nodes " + std::to_string(info.nodes);
-    if (info.score) res += " score cp " + std::to_string(*info.score);
+    // Mate display. If abs(score) > 20000 its a certain win (or loss).
+    // Length to mate (or tbwin) is then (+ or -) abs(score)-20000. 
+    if (info.score) {
+      if (abs(*info.score) >= 20000)
+        res +=
+            " score mate " +
+            std::to_string(*info.score + ((*info.score > 0) ? -20000 : 20000));
+      else
+        res += " score cp " + std::to_string(*info.score); 
+    }
     if (info.hashfull >= 0) res += " hashfull " + std::to_string(info.hashfull);
     if (info.nps >= 0) res += " nps " + std::to_string(info.nps);
     if (info.tb_hits >= 0) res += " tbhits " + std::to_string(info.tb_hits);
+    if (info.certain >= 0) res += " certain " + std::to_string(info.certain);
+    if (info.bounds >= 0) res += " bounds " + std::to_string(info.bounds);
```

Comment:
I think this might be breaking UCI protocol by adding non-standard keywords?

Reply:
Its common practice to add keywords, they should not really matter. But its more that i looked what other engines are doing - some are also outputting bounds and other info like this. Does it break a specific gui? The uci specification is rather vague on this. It works in arena, cutechess, chessbase... What gui gives you problems?

Reply:
I'm not aware of a specific GUI, but I remember "winrate" being removed from lczero for this reason: https://github.com/glinscott/leela-chess/issues/339.

Reply:
Yeah I've not seen any custom keywords in uci protocols. I only saw them after `string` (like we do with verbose move stats). I think it's better to remove them and output to the log only.

Also files in `chess/` subdirectory are attempted to be engine-independent, and adding custom MCTS fields there couples components.

Reply:
Will remove these counters.